module03/solution.py

```diff
@@ -0,0 +1,96 @@
+"""Solutions of module 3"""
+import math
+
+def sum_of_digits(num):
+    """Return sum of digits of an integer n."""
+    num = abs(num)
+    summ = 0
+    while num > 0:
+        summ += num%10
+        num //= 10
+    return summ
+
+def reverse_list(lst_orig):
+    """Return list in reverse order"""
+    lst_rev = []
+    while lst_orig:
+        lst_rev.append(lst_orig.pop())
+    return lst_rev
+
+def to_digits(num):
+    """Return a list of digits of an integer n"""
+    num = abs(num)
+    digits = []
+    while num > 0:
+        digits.append(num%10)
+        num //= 10
+    return reverse_list(digits)
+
+def to_number(digits):
+    """Return a number based on digits"""
+    digits = reverse_list(digits)
+    num = 0
+    for i, val in enumerate(digits):
+        num += val * 10 ** i
+    return num
+
+def count_vowels(strng):
+    """Return count ov vovels in a string"""
+    vowels = 'aeiouy'
+    num_vowels = 0
+    for char in strng.lower():
+        if char in vowels:
+            num_vowels += 1
+    return num_vowels
+
+def count_consonants(strng):
+    """Return count ov consonants in a string"""
+    consonants = 'bcdfghjklmnpqrstvwxz'
+    num_consonants = 0
+    for char in strng.lower():
+        if char in consonants:
+            num_consonants += 1
+    return num_consonants
+
+def prime_number(num):
```

Comment:
Clean and simple implementation. However, a relative simple speed up is to test up to **sqrt(num)** *square root*

Reply:
I've forgotten some important basic theory, sorry!